### PR TITLE
Spar Polysemy: Logger effect

### PR DESCRIPTION
**Pull request by isovector**
This PR pulls a Logger effect out of Spar. It's a little roundabout currently, for two reasons:

1. `saml2-web-sso`'s exported interface requires `HasLogger`, so we keep a `HasLogger` instance around.
2. `tinylog` doesn't export enough of its internals to abstract around its usage, so we require a `Logger (Msg -> Msg)` effect to be around

Point 1 can be cleaned up by either polysemizing `saml2-web-sso`, or adding an effect to wrap it. If we're planning to do the former (currently in scope for the polysemizing project), the latter is wasted work. It seems to me like this is worth punting on until we have a game plan. In the meantime, the logger effect is nice to have teased apart, and easy to change when we figure out the longer term plan.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/spar/src/Spar/Sem/Logger.hs
Comment (fisx) [reply services/spar/src/Spar/Sem/Logger.hs:39]: AHA!  :)

```
Spar/App.hs:159:14: warning: [-Wname-shadowing]
    This binding for ‘err’ shadows the existing binding
      imported from ‘Spar.Sem.Logger’ at Spar/App.hs:115:1-32
      (and originally defined at Spar/Sem/Logger.hs:39:1-3)
    |
159 |   throwError err = Spar $ throwError err
    |              ^^^
```

That's probably a bad small.  Import qualified or explicit, then this won't bite you.  :)

File: services/spar/src/Spar/App.hs
Comment (fisx) [reply services/spar/src/Spar/App.hs:165]: I just realized that that's true for all the many constraints of the functions exposed by saml2-web-sso.  Oh dear.  I'll try to hack around that.

File: services/spar/src/Spar/App.hs
Comment (fisx) [inline services/spar/src/Spar/App.hs:441]: I really want these `$` to become `.`.  I feel what you mean by "diffs are bad with ormolu" now...  :-)

File: services/spar/src/Spar/App.hs
Comment (fisx) [inline services/spar/src/Spar/App.hs:442]: I would move `mapLogger` into the interpreter (or wrapping a more convenient interpreter around the one you have, of course).  Then this here would be a little less mind-bending.

**Review by fisx — APPROVED** [review]
Good to go!  Iff you agree with any of my suggestions, please act on them and merge without another review.

File: services/spar/src/Spar/App.hs
Comment (isovector) [reply services/spar/src/Spar/App.hs:165]: This instance disappears in my new branch (no PR yet), so going to hold off on this comment.

File: services/spar/src/Spar/API.hs
Comment (isovector) [reply services/spar/src/Spar/API.hs:328]: We can, but I'll leave a TODO because the original code is using a string (via `HasLogger.logger`).

**Pull request merged**
